src/openrct2/object/Object.h

```diff
@@ -81,8 +81,14 @@ typedef struct rct_object_entry {
         uint8 end_flag; // needed not to read past allocated buffer.
         uint32 flags;
     };
-    char name[8];
-    uint32 checksum;
+    union {
+        char nameWOC[9];
+        struct {
+            char name[8];
+            uint32 checksum;
```

Comment:
Hmm… Unifying a `char` (one byte) with a `uint32` (four bytes)? This doesn't seem quite right.

Reply:
Just look at the usages. There are some places where no checksum is needed (or read, even).

Reply:
I saying the amount of unified bytes doesn't match. You're implicitly depending on alignment for the unification to work. I'm not sure how that's specified in the C++17 spec, but it's tricky at best.

Reply:
Just to clarify further, it would make more sense, to me, to instead unify the two as:
```diff
-    char name[8];
-    uint32 checksum;
+    union {
+        char nameWOC[12];
+        struct {
+            char name[8];
+            uint32 checksum;
+        };
+    };
```

This way, you avoid the trouble with alignment or endianness.

Reply:
Right, changed it.